content/immutable-owner.md

```diff
@@ -0,0 +1,393 @@
+```
+
+If you decide to use devnet, and have issues with airdropping SOL. Feel free to add the `keypairPath` parameter to `initializeKeypair`. You can get this by running `solana config get` in your terminal. And then go to [faucet.solana.com](https://faucet.solana.com/) and airdrop some SOL to your address. You can get your address by running `solana address` in your terminal.
+
+### 3. Helpers
+
+When we pasted the `index.ts` code from earlier, we added the following helpers:
+
+- `initializeKeypair`: This function creates the keypair for the `payer` and also airdrops 2 testnet SOL to it
+- `makeKeypairs`: This function creates keypairs without airdropping any SOL
+
+Additionally, we have some initial accounts:
+  - `payer`: Used to pay for and be the authority for everything
+  - `mintKeypair`: Our mint
+  - `ourTokenAccountKeypair`: The token account owned by the payer that we'll use for testing
+  - `otherOwner`: The token account we'll try to transfer ownership of the two immutable accounts to
+
+### 4. Create mint
+
+Let's create the mint we'll be using for our token accounts.
+
+Inside of `src/index.ts`, the required dependencies will already be imported, along with the aforementioned accounts. Add the following `createMint` function beneath the existing code:
+
+```tsx
+// CREATE MINT
+const mint = await createMint(
+  connection,
+  payer,
+  mintKeypair.publicKey,
+  null,
+  2,
+  undefined,
+  undefined,
+  TOKEN_2022_PROGRAM_ID,
+);
+```
+
+### 5. Create Token Account with immutable owner
+
+Remember all ATA's come with the `immutable owner` extension. However, we're going to create a token account using a keypair. This requires us to create the account, initialize the immutable owner extension, and initialize the account. 
+
+Inside the `src` directory, create a new file named `token-helper.ts` and create a new function within it called `createTokenAccountWithImmutableOwner`. This function is where we'll be creating the associated token account with the immutable owner. The function will take the following arguments:
+
+- `connection`: The connection object
+- `mint`: Public key for the new mint
+- `payer`: Payer for the transaction
+- `owner`: Owner of the associated token account
+- `tokenAccountKeypair`: The token account keypair associated with the token account
+
+```ts
+import { ExtensionType, TOKEN_2022_PROGRAM_ID, createInitializeAccountInstruction, createInitializeImmutableOwnerInstruction, getAccountLen } from "@solana/spl-token";
+import { Connection, Keypair, PublicKey, SystemProgram, Transaction, sendAndConfirmTransaction } from "@solana/web3.js";
+
+export async function createTokenAccountWithImmutableOwner(
+  connection: Connection,
+  mint: PublicKey,
+  payer: Keypair,
+  owner: Keypair,
+  tokenAccountKeypair: Keypair
+): Promise<string> {
+
+  // CREATE ACCOUNT INSTRUCTION
+
+  // ENABLE IMMUTABLE OWNER INSTRUCTION
+
+  // INITIALIZE ACCOUNT INSTRUCTION
+
+  // SEND TO BLOCKCHAIN
```

Comment:
This is a new style that's distinct from the rest of the course. I'd just use `createTokenAccountWithImmutableOwner` then state in text that this function will: 

- Make an instruction to create an account
- Make an instruction to enable 'immutable owner'
- Make an instruction to initialize the account
- Add these instructions to a transaction and send them to the blockchain.

Reply:
Fixed in #388